mlflow/utils/async_logging/run_batch.py

```diff
@@ -9,27 +9,28 @@
 class RunBatch:
     def __init__(
         self,
-        run_id: str,
-        params: List[Param],
-        tags: List[RunTag],
-        metrics: List[Metric],
-        completion_event: threading.Event,
-    ) -> None:
+        run_id: Optional[str] = None,
+        params: Optional[List["Param"]] = None,
+        tags: Optional[List["RunTag"]] = None,
+        metrics: Optional[List["Metric"]] = None,
```

Comment:
```suggestion
        params:List["Param"] = [],
        tags:List["RunTag"] = [],
        metrics:List["Metric"] = [],
```
It seems the batch merging logic doesn't handle null for params/tags/metrics. For example, ` merged_batch.params.extend(batch.params)` will raise exception when params is None. Shall we make it non-null and default to an empty list instead?